### PR TITLE
deprecate assert package

### LINKED ISSUE #5672: migrate to @endo/errors package
## What is the Problem Being Solved?

`assert` is a global in SES. However, IDEs don't know about it or its members. To allow automatic imports we have `@agoric/assert`, but it's useful outside of Agoric.

## Description of the Design

Move most of `@agoric/assert` to use `@endo/errors`. Move `NonNullish` checked cast to `@agoric/internal`.

# Acceptance

No more use of global `assert` in agoric-sdk. This will support getting assertion functions without having set up a SES environment.

### PR DESCRIPTION
**Pull request by dckc**
fixes #5672

see also #3436

### CODE REVIEW

**Review by erights — APPROVED** [review]
LGTM thanks!

See also https://github.com/Agoric/agoric-sdk/issues/5605

**Review by turadg — CHANGES_REQUESTED** [review]
This change is fine but I don't think meets the bar to close #5672 .

I think we should also,
1. Remove all imports from this package in agoric-sdk
2. [Deprecate the package in NPM](https://docs.npmjs.com/cli/v8/commands/npm-deprecate)

We could also consider using the `@deprecated` jsdoc comment on the exports of the package, but I don't think that's useful if we do the two above.

I'm willing to do that as part of https://github.com/Agoric/agoric-sdk/issues/5605 . I pushed some work towards that into this branch.

Fwiw, I think the `const` form is worse because even though `assert` is on global very common uses of it repeat declarations like `const { details: X } = assert;`. That being such a consistent and supported pattern, I'd expect `import { X } from '@agoric/assert';`. Having that export would let IDEs autocomplete when someone tries to use X as a function.

So second thought, I'd like to request that we don't deprecate this package and instead explain why it exists. I'll leave the above to show you how I got here :)

**Comment by dckc** [comment]
> ... I'd like to request that we don't deprecate this package ...

That seems like a disagreement about the issue (#5672), so I withdraw this PR.

File: packages/ERTP/test/swingsetTests/basicFunctionality/bootstrap.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { E } from '@endo/eventual-send';
-import { assert, details as X } from '@agoric/assert';
+const { details: X } = assert;
```
Comment (erights) [inline packages/ERTP/test/swingsetTests/basicFunctionality/bootstrap.js:2]: Not sure why lint is correctly flagging some of these as needing to be moved to after the imports, but is not flagging others, such as this one. In any case, all of these need to be after the imports.

**Review by erights — COMMENTED** [review]
Is Closing this PR the right thing? It must have been a substantial amount of manual work, and would all be relevant again if we do end up deciding to deprecate the import. Please let's not lose this.

**Comment by turadg** [comment]
> It must have been a substantial amount of manual work

maybe 10min? mostly global find-replace.

> let's not lose this

I've noted this in the issue for when it gets picked up again.

**Comment by erights** [comment]
Thanks!

**Pull request closed without merging**